### DISCUSSION TITLE
Use all available CPUs with PM2

### DISCUSSION THREAD

**Discussion by El-Tommy** [socketio/socket.io#4599] [Q&A] [answered]
I have a virtual machine with 4 CPUs, but as by default NodeJS only runs on 1 CPU, I wanted to take full advantage of this VM's potential to run my server with Socket.IO.

I saw [this article](https://socket.io/docs/v4/pm2/) in the Socket.IO documentation. If I use it, can I use all the CPUs and make the users, regardless of the room+CPU, able to communicate with each other?

**Comment by qiulang** [comment, marked as answer]
Yes as document says 

> Please note that if you have several hosts each running a PM2 cluster, you will have to use another adapter, like the [Redis adapter](https://socket.io/docs/v4/redis-adapter/).

**Reply by El-Tommy** [reply]
What if it is necessary to use it with HTTPs? How I do?

I noticed that when setting a port manually with `httpsServer.listen(3001);` the code doesn't work and the client keeps trying to reconnect.

**Reply by darrachequesne** [reply]
@El-Tommy please see my answer [here](https://github.com/socketio/socket.io/discussions/4603#discussioncomment-4743653).
